selfapi.js

```diff
@@ -367,6 +372,21 @@ API.prototype = {
     });
   },
 
+  // Build index routes.
+  toAPIIndex: function (basePath) {
+    var fullPath = normalizePath(this.path, basePath) || '/';
+    return function () {
```

Comment:
Oh, you're returning a function? Why not simply compute the `routes` object and return it, making `toAPIIndex` a very simple exporter, instead of a function-returning function?